Slenbder/catalog1 3

slenbder opened the pull request:
**Способ вёрстки: код**

**Архитектура: MVVM**

## Выполнена работа по модулю 1: Основная верстка и базовая функциональность

### Выполненные задачи:

- **Изучение технического задания и макетов**
    
    - _Описание:_ Ознакомиться с требованиями проекта и макетами экранов.
    - _Оценка:_ **1 час**; _Факт:_ 1 час.
    
- **Создание структуры проекта в Xcode**
    
    - _Описание:_ Настроить новый проект, настроить архитектуру MVVM, создать необходимые группы и файлы.
    - _Оценка:_ **2 часа**; _Факт:_ 2 часа.
    
- **Верстка экрана каталога (`UITableView`)**
    
    - _Описание:_
        - Создать `CatalogViewController` и соответствующий `CatalogViewModel`.
        - Добавить `UITableView` для отображения коллекций.
        - Создать кастомную ячейку `CollectionTableViewCell` с обложкой, названием и количеством NFT.
        - Настроить Auto Layout и UI-элементы согласно макету.
    - _Оценка:_ **6 часов**; _Факт:_ 5 часов.
    
- **Верстка экрана коллекции NFT (`UICollectionView`)**
    
    - _Описание:_
        - Создать `CollectionViewController` и соответствующий `CollectionViewModel`.
        - Добавить `UICollectionView` для отображения NFT.
        - Создать кастомную ячейку `NFTCollectionViewCell` с изображением, названием, рейтингом, стоимостью, кнопками избранного и корзины.
        - Настроить Auto Layout и UI-элементы согласно макету.
    - _Оценка:_ **6 часов**; _Факт:_ 5 часов.
    
- **Подготовка к ревью**
    
    - _Описание:_ Проверить код на соответствие стандартам, провести рефакторинг при необходимости, написать комментарии.
    - _Оценка:_ **1 час**; _Факт:_ 1 час.

---

usvladis committed:
Проведена первичная подготовка проекта к работе
-Добавлена локализация
-Добавлены цвета и иконки
-Создан общий таб бар

zhuk76787 committed:
удалил StoryBoard, исправил ошибку в TabBarViewController, добавил иконку приложения

zhuk76787 committed:
добавил Preview

zhuk76787 committed:
оптимизировал TabBarController

zhuk76787 committed:
добавил иконки монеток

zhuk76787 committed:
сделал custom launchScreen и перераспределил картинки с монетками

slenbder committed:
Верстка экрана каталога

- Создан CatalogViewController и соответствующий CatalogViewModel.
- Добавлен UITableView для отображения коллекций.
- Создана кастомная ячейка CollectionTableViewCell с обложкой, названием и количеством NFT.
- Настроены Auto Layout и UI-элементы согласно макету.

slenbder committed:
Верстка экрана коллекции NFT

- Создал CollectionViewController и соответствующий CollectionViewModel.
- Добавил UICollectionView для отображения NFT.
- Создал кастомную ячейку NFTCollectionViewCell с изображением, названием, рейтингом, стоимостью, кнопками избранного и корзины.
- Настроил Auto Layout и UI-элементы согласно макету.

slenbder committed:
Мелкие правки и багфиксы:

- Увеличил расстояние между ячейками каталога
- Изменил направление отображения коллекции (сбоку вылетает)
- Имя автора теперь кнопка, которая будет переводить на WebView

slenbder committed:
Мелкие правки 2:

- Кнопка сортировки теперь является элементом NavifationBar
- На экране коллекции теперь все элементы в ScrollView
- Реализована функция динамического обновления высоты коллекции

slenbder committed:
Изменил файл декомпозиции - прописал часы

Kirchberg commented:
Пожалуйста, не создавай новые Pull Requests для каждого изменения. Все последующие правки и дополнения вноси в уже существующий PR. Это облегчит процесс ревью и слежения за изменениями. Просто обновляй PR, добавляя новые коммиты по мере необходимости 🙏

slenbder commented:
> Пожалуйста, не создавай новые Pull Requests для каждого изменения. Все последующие правки и дополнения вноси в уже существующий PR. Это облегчит процесс ревью и слежения за изменениями. Просто обновляй PR, добавляя новые коммиты по мере необходимости 🙏

Да, я поздновато сообразил что так можно не делать. Раз уж создал новый, тогда уже в него все буду подгружать

slenbder committed:
Исправления в ПР к Модулю 1:

- Упростил сущность кнопки сортировки (перенес из метода в жизненный цикл), исправил опечатку, перенес структуру Preview в конец файла.
- Исправил `AlertController`. Сделал его финальным и упростил название.
- `CatalogViewController` теперь называется `CatalogDetailsScreenViewController`.
- В расширении к CatalogDetailsScreenViewController заменил принт ошибки на assertionFailure.
- Упростил метод `updateRating`.
- В файле `CatalogTableViewCell` упорядочил методы.

slenbder committed:
Исправления в ПР к Модулю 1:

- Упростил сущность кнопки сортировки (перенес из метода в жизненный цикл), исправил опечатку, перенес структуру Preview в конец файла.
- Исправил `AlertController`. Сделал его финальным и упростил название.
- `CatalogViewController` теперь называется `CatalogDetailsScreenViewController`.
- В расширении к CatalogDetailsScreenViewController заменил принт ошибки на assertionFailure.
- Упростил метод `updateRating`.
- В файле `CatalogTableViewCell` упорядочил методы.

slenbder committed:
Исправления в ПР (2) к Модулю 1:

- Исправил отступ на экране каталога, по совету наставника вынес rightBarButtonItem переменной и добавил в отдельный метод
- Немного поработал над темной темой экрана (не до конца)

slenbder committed:
Исправления в ПР (2) к Модулю 1:

- Исправил отступ на экране каталога
- Немного поработал над темной темой экрана (не до конца)

slenbder committed:
Исправления в ПР (2) к Модулю 1:

- Доработал темную тему экрана Каталога

slenbder committed:
Merge remote-tracking branch 'refs/remotes/origin/slenbder/catalog1-3'

#Conflicts:
#	FakeNFT/Scenes /Catalog/View/CatalogViewController.swift
